### PR TITLE
Port metrics tests

### PR DESCRIPTION
**Pull request by cquiroz**
Signed-off-by: Carlos Quiroz <carlos.m.quiroz@gmail.com>

### CODE REVIEW

**Comment by cquiroz** [comment]
I undid the deletion of the old Matchers files as it breaks mima compatibility.

**Review by rossabaker — APPROVED** [review]
Which matchers?  `IOMatchers` or `Http4sMatchers`?

**Comment by cquiroz** [comment]
I deprecated `Http4sLegacyMatchers` and `Http4sLegacyMatchersIO`

File: testing/src/test/scala/org/http4s/testing/Http4sLegacyMatchers.scala [hunk 1]
```diff
@@ -24,6 +24,7 @@ import org.http4s.headers._
 import org.http4s.util.CaseInsensitiveString
 import org.specs2.matcher.{RunTimedMatchers => Specs2RunTimedMatchers, _}
 
+@deprecated("Use Http4sSuite and munit-cats-effect-instead", "0.21.6")
```
Comment (rossabaker) [inline testing/src/test/scala/org/http4s/testing/Http4sLegacyMatchers.scala:27]: Wait, these are in `src/test`.  We should be able to just eliminate them, no?

Also, next one is 0.21.16, not 0.21.6.

Comment (cquiroz) [reply testing/src/test/scala/org/http4s/testing/Http4sLegacyMatchers.scala:27]: I'd be happy to delete them 😀

**Pull request merged**